splunk_connect_for_snmp_traps/manager/trap_server.py

```diff
@@ -191,24 +199,24 @@ def snmp_callback_function(
             hostname, aliaslist, ipaddrlist = socket.gethostbyaddr(device_ip)
             parts = str(hostname).split(".")
             name = parts[0]
-            # print(name)
+
             if len(parts) > 1:
                 header["Agent_Hostname"] = name
                 logger.debug(f"host={header['Agent_Hostname']} device_ip={device_ip}")
             else:
                 header["Agent_Hostname"] = device_ip
                 logger.debug(f"device_ip={device_ip}")
-        except:
+        except socket.herror:
             logger.debug(f"device_ip={device_ip}")
             header["Agent_Hostname"] = device_ip
             pass
 
         header["Agent_Address"] = device_ip
 
-        # Send API call to SNMP MIB server to get var_binds translated
-        # mib_server_url = self._server_config["snmp"]["mib-server"]["url"]
-        mib_server_url = os.environ["MIBS_SERVER_URL"]
-        trap_event_string = get_translation(var_binds, mib_server_url)
+        loop = asyncio.get_event_loop()
+        tasks = get_translation(var_binds, os.environ["MIBS_SERVER_URL"])
```

Comment:
get_translation is async and we are in a sync part of code.  So calling it returns some kind of hook that we are waiting for in asyncio.gather, right?

Reply:
From docs:
run_until_complete """Run the event loop until a Future is done.

gather: """Return a future aggregating results from the given coroutines/futures.

Maybe gather is not necessary, I got it from some code examples and there it was used for multiple coroutines and here we have just one so maybe loop.run_until_complete(tasks) would be enough